### ISSUE TITLE
Q: Annotation vs Attributes: different encoding field implementation?

### ISSUE THREAD

**Issue by shandyDev** [zircote/swagger-php#1210]
Hello

From you reference documentation we can see (https://zircote.github.io/swagger-php/reference/annotations.html#mediatype):

>    encoding : array
>     A map between a property name and its encoding information.
>     The key, being the property name, must exist in the schema as a property.
>     The encoding object shall only apply to requestBody objects when the media type is multipart or
>     application/x-www-form-urlencoded.

BUT there with attributes https://zircote.github.io/swagger-php/reference/attributes.html#mediatype

>  encoding : string|null
>     No details available.

So question is it bug or some special purpose?

https://github.com/zircote/swagger-php/blob/master/src/Annotations/AbstractAnnotation.php#L116-L124

So with attributes code `if (is_array($value)) {` not pass this IF. And here annotations can not move to attributes properly

```
     *     @OA\RequestBody(
     *          required=true,
     *          @OA\MediaType(
     *             mediaType="multipart/form-data",
     *             encoding={
     *                 "imageFiles": {"contentType": "image/png, image/jpeg, image/webp"},
     *             },
     *             @OA\Schema(
     *                 required={"imageFiles"},
     *                 @OA\Property(property="imageFiles", type="array", @OA\Items(type="string"), format="binary"),
     *             )
     *         )
     *     ),
```

**Comment by DerManoMann** [comment]
Looks like a bug to me. Keeping things in sync is a pain; I thought we finally got things right but perhaps there is still value in having some code check that annotations and attributes are compatible.

**Comment by shandyDev** [comment]
We now moving our project from annotation to attributes, so until this all was fine.)
But maybe yes, need write some integration tests.
Also we can make PR for this bug.

**Comment by DerManoMann** [comment]
PRs are always appreciated.

**Comment by DerManoMann** [comment]
Fixed via #1217
